protocols/floodsub/src/layer.rs

```diff
@@ -216,6 +216,25 @@ where
         propagation_source: PeerId,
         event: FloodsubRpc,
     ) {
+        // Update connected peers topics
+        for subscription in event.subscriptions {
+            let mut remote_peer_topics = self.connected_peers
+                .get_mut(&propagation_source)
+                .expect("This peer just sent us a message");
```

Comment:
q: not sure if this is ok. Is it possible for a peer to be removed from `connected_peers` at this point?

Reply:
The code is good. The swarm guarantees that we can't receive events from disconnected nodes.
However the proof should be a bit more extended:
```suggestion
                .expect("connected_peers is kept in sync with the peers we are connected to ; we are guaranteed to only receive events from connected peers ; qed");
```